beamer/config.py

```diff
@@ -124,8 +125,16 @@ def load(config_path: Path, options: dict[str, Any]) -> Config:
     password = _get_value(config, "account.password")
     account = account_from_keyfile(path, password)
 
+    source_w3 = Web3(HTTPProvider(source_rpc_url, request_kwargs=dict(timeout=5)))
+    source_chain_id = ChainId(source_w3.eth.chain_id)
+
+    target_w3 = Web3(HTTPProvider(target_rpc_url, request_kwargs=dict(timeout=5)))
+    target_chain_id = ChainId(target_w3.eth.chain_id)
+
     deployment_info = load_deployment_info(Path(config["deployment-dir"]))
-    token_match_checker = TokenMatchChecker(list(config["tokens"].values()))
+    token_match_checker = TokenMatchChecker(
+        list(config["tokens"].values()), source_chain_id, target_chain_id
+    )
     return Config(
         account=account,
         deployment_info=deployment_info,
```

Comment:
I am going to comment here, but it goes for the whole commit/PR: I don't think it is a good idea to do this check via `TokenMatchChecker`.

Firstly, as can be seen above, we have network traffic during configuration load. It also mixes mechanism (loading of configuration) with policy (filtering). We should not do that.

Secondly, I feel the current token match checker is simply not a good fit for the new filter. Maybe we should consider renaming it? The issues mentions "listening wrong target chain request", which leads me to think we should rather consider a different way to reject requests that we are not interesting in. Personally, I would rather see an `if` check somewhere than push this check into token match checker as is. (This only under the assumption that there are not many places for such a check, naturally.)

Reply:
Thanks for the comment.
In my view, we should revise or refactor respectively, the TokenMatchChecker into a module which filters for a certain set of conditions.
We have the issue that with adding a third rollup, we increased one dimension of potential filters. Currently we only filtered if events are incorrect semantically (wrong addresses/tokenpairs, etc). 
Now we have the fact that we created a new dimension of "valid" events but not valid for this particular agent instance.
This creates the need of filtering out events which are not of interest for the agent. For each event and therefore each event handler, we need to check whether we should filter this event because it is irrelevant for the agent.

If we dont do this then two things can happen:
a) the agent eventually crashes (this is what we saw and why this issue is here)
b) the event starves in the event loop.

We are lucky that most Claim related events will be filtered out naturally by the fact that we dont create a request object. But for the future, we need to have a more systematic way of filtering unwanted events as we may introduce new events by the contracts.

IMO the TokenMatchChecker is doomed to be replaced by something much more powerful, a generalized module which checks if this event is ok to be processed.

The real question you are raising here (besides the technical opinions like having network traffic during configuration load) is if we should assume that a token pair is valid if it belongs two a chain which the agent doesn't know or should it raise a invalid token pair error. In the past, a chain id other than the two the agent is running on would probably have failed because the agent doesn't know about this other chain.

Since we are now using one agent config for all agent instances we have more than 2 chains per token in the config which are considered valid. The TokenMatchChecker does not catch this issue. IMO it is the purpose of the TokenMatchChecker of each agent instance to check the `(chain_id, token_address)` tuple if this is a valid token pair. Now, we should ask ourselves if it is a valid token pair even though the agent is not running and therefore doesnt know anything about this chain. In the past the TokenMatchChecker would have noticed this and filtered out the event. 

In general, I think it is totally fine and it's the purpose of the TokenMatchChecker to detect a request which is not a valid pair for this particular agent instance. 

Nonetheless, for a long time I want to rework the way we validate events if they are of interest for the agent instance, because I also think having tons of if statements here and there in the event handlers is also not a good design for the future. 
As a result I don't see this as a change for long as I think, as mentioned, that we need to replace the TokenMatchChecker with a more generalized way of analyzing events.

Reply:
hmm let's look from a different angle on this issue, maybe this is a better approach.

Currently, an agent runs on two roll ups, A and B. So the agent is only concerned about information of A and B. And I strongly think that the agent should only consume information about these roll ups. 
Now, we started to create a config file which also contains information of roll up C. We created - let's call it - a multi instance config.
We now have the conflict that each agent instance will consume information of rollups, it doesnt know. The reason why I think we should avoid storing information of roll up C in the A->B agent instance is that the agent cannot possibly know even the existence of this roll up.
And as we have seen, things break if we do this. So by creating a multi instance agent config, I see the need of filtering necessary information for each of the instances. IOW, Agent A->B should only care and consume/store the information of A and B and not of C.

This is basically what is done in the TokenMatchChecker by dropping the token information of roll up C. This could also be done on a higher level but I would definetly not want to have information of roll up C inside of A->B.
By filtering out the token information of roll up C, it will naturally drop any request which is not A->B. Since I would recommend to do this anyway, there is no need of a specific if statement inside of the RequestCreated handler. The filtering comes for free.

I'm open for suggestions of a different way how to filter the config for A->B information, if you think this isn't the right place.

All this being said, it shows that this change is not here for long as we will collapse all agent instances into one agent process. With the planned changes of our agent software, the filtering process will need to be implemented properly in order to build the foundation for a scalable agent software.

Reply:
Thank you for elaborating on the background for these changes, I really appreciate it. And I do see where you are coming from and agree for the most part. And yes, you are fully on-point regarding the multi-chain instances/config.

My concern here is rather technical and quite specific. After having some more look into the code, I think
removing the unsupported tokens (pairs of `chain_id, token_address`) from the loaded config just after
```
config = beamer.config.load(config_path, options)
```
in `beamer.cli.main` would be the best way to go, given circumstances. The token match checker would then stay as-is, and only the `cli` module which is tasked with setting up the concrete agent instance can do this fiddling, query chain ID and remove those tokens we do not need. I think that'd be a much cleaner solution for now. (And of course, fully agreed that this will change in the future.)

What do you think?

Reply:
Sounds like a good idea. Let me change it.